bots/notify/src/main/java/org/openjdk/skara/bots/notify/RepositoryWorkItem.java

```diff
@@ -361,16 +365,31 @@ public Collection<WorkItem> run(Path scratchPath) {
             }
 
             boolean hasBranchHistory = !history.isEmpty();
+            int existingCommits = localRepo.getExistingCommits();
             for (var ref : knownRefs) {
                 if (!hasBranchHistory) {
                     log.warning("No previous history found for any branch - resetting mark for '" + ref.name());
-                    for (var listener : listeners) {
-                        log.info("Resetting mark for branch '" + ref.name() + "' for listener '" + listener.name() + "'");
-                        history.setBranchHash(new Branch(ref.name()), listener.name(), ref.hash());
+                    if (existingCommits <= THRESHOLD) {
+                        // In this case, the repo will be considered as a new repo, notify bot will notify start from the first commit
+                        log.info("This is a new repo");
```

Comment:
```suggestion
                        log.info("This is a new repo, starting notifications from the very first commit");
```
This that log message, I think the comment above becomes redundant.